api/src/routes/identity/index.test.ts

```diff
@@ -10,477 +10,404 @@ import { UserIdentityMock } from '../../test/mocks/identities';
 		sendStatusMock = jest.fn();
 		nextMock = jest.fn();
-		userService = new UserService({} as any, '');
-		const authorizationService = new AuthorizationService(userService);
-		userRoutes = new IdentityRoutes(userService, authorizationService);
-
-		res = {
-			send: sendMock,
-			sendStatus: sendStatusMock
-		};
-	});
 
-	it('should call searchUser with expected user search', async () => {
-		const expectedUserSearch: UserSearch = {
-			type: UserType.Person,
-			index: 1,
-			limit: 1,
-			username: 'charlie',
-			verified: true,
-			organization: 'IOTA',
-			registrationDate: getDateFromString('2021-02-12T14:58:05+01:00')
-		};
-		const searchUserSpy = spyOn(UserDb, 'searchUsers').and.returnValue([]);
-		const req: any = {
-			params: {},
-			query: {
-				type: 'Person',
-				'registration-date': '2021-02-12T14:58:05+01:00',
-				verified: 'true',
-				limit: '1',
-				index: '1',
-				username: 'charlie',
-				organization: 'IOTA',
-				'subscribed-channel-ids': ['test-address', 'test-address2']
-			},
-			body: null
+		const identityConfig: IdentityConfig = {
+			keyCollectionTag: 'key-collection',
+			explorer: '',
+			network: 'test',
+			node: '',
+			keyType: 0,
+			hashFunction: 0,
+			hashEncoding: 'base58'
 		};
-
-		await userRoutes.searchUsers(req, res, nextMock);
-
-		expect(searchUserSpy).toHaveBeenCalledWith(expectedUserSearch);
-	});
-});
-
-describe('test GET user', () => {
-	let sendMock: any, sendStatusMock: any, nextMock: any, res: any, userService: UserService, userRoutes: IdentityRoutes;
-	beforeEach(() => {
-		sendMock = jest.fn();
-		sendStatusMock = jest.fn();
-		nextMock = jest.fn();
-		userService = new UserService({} as any, '');
+		ssiService = SsiService.getInstance(identityConfig);
+		userService = new UserService(ssiService as any, serverSecret);
 		const authorizationService = new AuthorizationService(userService);
 		userRoutes = new IdentityRoutes(userService, authorizationService);
 
 		res = {
 			send: sendMock,
-			sendStatus: sendStatusMock
+			sendStatus: sendStatusMock,
+			status: jest.fn(() => res)
 		};
 	});
-
-	it('should return bad request if no user id is given as parameter', async () => {
-		const req: any = {
-			params: {},
-			body: null
-		};
-
-		await userRoutes.getUser(req, res, nextMock);
-		expect(sendStatusMock).toHaveBeenCalledWith(400);
+	describe('test search route', () => {
+		it('should call searchUser with expected user search', async () => {
+			const expectedUserSearch: UserSearch = {
+				type: UserType.Person,
+				index: 1,
+				limit: 1,
+				username: 'charlie',
+				verified: true,
+				organization: 'IOTA',
+				registrationDate: getDateFromString('2021-02-12T14:58:05+01:00')
+			};
+			const searchUserSpy = spyOn(UserDb, 'searchUsers').and.returnValue([]);
+			const req: any = {
+				params: {},
+				query: {
+					type: 'Person',
+					'registration-date': '2021-02-12T14:58:05+01:00',
+					verified: 'true',
+					limit: '1',
+					index: '1',
```

Comment:
should the two values for limit and index be strings or numbers?

Reply:
since it is a query param string is fine, this will be parsed as a number